Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1701,6 +1701,35 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     fireActionClickedEventIfNeeded(tab);
 }
 
+const WebExtensionContext::CommandsVector& WebExtensionContext::commands()
+{
+    if (m_populatedCommands)
+        return m_commands;
+
+    for (auto& data : extension().commands())
+        m_commands.append(WebExtensionCommand::create(*this, data));
```

Comment:
Since `m_commands` is uninitialized here, and we don't hint at the size, it could be expensive to loop and append (if there are a lot of commands -- maybe there never are).

Still it's simple to write the code this way and avoid the risk entirely:

I think `extension().commands()` returns a copy, so you can use a move operator into the lambda. If that's wrong, just make it a reference:

```
m_commands = WTF::map(extension().commands(), [&](auto&& data) -> Ref<WebExtensionCommand> {
    return WebExtensionCommand::create(*this, data);
|);
```